pkg/io/v2/opener.go

```diff
@@ -0,0 +1,363 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	"gocloud.dev/blob"
+	"gocloud.dev/gcerrors"
+
+	"k8s.io/test-infra/pkg/io/v2/providers"
+	"k8s.io/test-infra/prow/errorutil"
+)
+
+// Aliases to types in the standard library
+type (
+	ReadCloser  = io.ReadCloser
+	WriteCloser = io.WriteCloser
+)
+
+// Opener has methods to read, write paths, create signed URLs,...
+type Opener interface {
```

Comment:
All the consumers of this interface currently use `Reader` and `Writer` only, right? Can we keep it to just those two until we have a consumer that needs more (Presumably never going to happen if status-reconciler, gerrit and tide are the only users of this)?

Reply:
Yup. I'll remove the other (now) unused methods and reintroduce them in the next PRs